New PWA app icon how-to page (#26583)
mdn/content: files/en-us/web/progressive_web_apps/how_to/define_app_icons/index.md

For example, Windows can display your app icon as a 44x44 pixels image in the taskbar, or as a 150x150 pixels image in the start menu. Use the links below for more information about the icon sizes used by different operating systems:

- For Windows, see [Define icons and a theme color](https://learn.microsoft.com/microsoft-edge/progressive-web-apps-chromium/how-to/icon-theme-color) on learn.microsoft.com.
- For Android, see [Google Play icon design specifications](https://developer.android.com/distribute/google-play/resources/icon-design-specifications) on developer.android.com.
- For iOS and macOS, see [App icons](https://developer.apple.com/design/human-interface-guidelines/app-icons#App-icon-sizes) on developer.apple.com.

The sizes at which your icon is displayed depend on the operating system and may change over time. It's best to test your icon on all the devices and operating systems you want to support, and generate the sizes and image types that lead to the best results. You can also use a tool like [PWA Image Generator](https://www.pwabuilder.com/imageGenerator) to generate the necessary icon sizes for you from a single high-definition image.

> Note that some operating systems support [SVG](/en-US/docs/Web/SVG) icons which can help reduce the number of images you need to create since SVG can be scaled to any size automatically. However, there may be cases where your SVG icon doesn't scale down to small sizes nicely. To support all OSs and small sizes, create PNG versions of your icon as well. Learn more about how to reference the various icon sizes in [Reference icons in the web app manifest](#reference-icons-in-the-web-app-manifest).

Author: Thanks for the suggestions so far, I'm applying them locally and will push a new commit soon.

As for changing some to most, I don't know if most OS do support SVG for PWA icons. I thought I'd stay on the safe side and use "some".

![The Android 13 home screen, showing circular and color themed app icons](./android-13-home-screen.png)

Your PWA app icon should specifically support masking to look well integrated with operating systems that apply masks. Icons that do not support masking may appear cropped or smaller than expected. On Android, non-maskable icons are centered within the circular mask and given a white background, which may not look good with your icon.

The following image illustrates the difference between a maskable and a non-maskable icon on Android:

![A non-maskable icon on the left, as a small square within the app icon circle. A maskable icon on the right, filling in the entire app icon circle](./maskable-icon-comparison.png)

To start making your app icon maskable, use the `purpose` property in the icon objects of your web app manifest, and set its value to `maskable`. For example:

```json
{
  "icons": [
    {
      "src": "icon-192-maskable.png",
      "sizes": "192x192",
      "type": "image/png",
      "purpose": "maskable"
    }
  ]
}
```

> Using the `purpose` property is only the first step in making your icon maskable. You also need to make sure your icon looks well integrated with the host operating system by ensuring the important parts of the icon appear well within the mask's _safe zone_.

Reviewer: ```suggestion
Using the `purpose` property is just the last step in making your icon maskable. You need to start by making sure your icon looks well integrated with the host operating system by ensuring the important parts of the icon appear well within the mask's _safe zone_.
```
totally optional ;)
